File: example/src/main/java/com/jiangcm/baselib/guideView/component/LottieComponent.java
package com.jiangcm.baselib.guideView.component;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.Toast;

import com.jiangcm.base.widght.guideview.Component;
import com.jiangcm.baselib.R;

public class LottieComponent implements Component {

  @Override public View getView(LayoutInflater inflater) {

    LinearLayout ll = (LinearLayout) inflater.inflate(R.layout.layer_lottie, null);
    ll.setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View view) {
        Toast.makeText(view.getContext(), "引导层被点击了", Toast.LENGTH_SHORT).show();
      }
    });
    return ll;
  }

  @Override public int getAnchor() {
    return Component.ANCHOR_TOP;
  }

  @Override public int getFitPosition() {
    return Component.FIT_CENTER;
  }

  @Override public int getXOffset() {
    return 0;
  }

  @Override public int getYOffset() {
    return -30;
  }
}

File: example/src/main/java/com/jiangcm/baselib/ui/guideView/component/LottieComponent.kt
package com.jiangcm.baselib.ui.guideView.component

import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import android.widget.Toast
import com.jiangcm.base.widght.guideview.Component
import com.jiangcm.baselib.R

class LottieComponent : Component {

    var listener: (() -> Unit?)? = null

    fun setOnListener(listener: () -> Unit?) {
        this.listener = listener
    }


    override fun getView(inflater: LayoutInflater?): View {
        val ll = inflater?.inflate(R.layout.layer_lottie, null) as LinearLayout
        ll.setOnClickListener { view ->
            Toast.makeText(view.context, "引导层被点击了", Toast.LENGTH_SHORT).show()
        }
        return ll
    }

    override fun getAnchor(): Int {
        return Component.ANCHOR_TOP
    }

    override fun getFitPosition(): Int {
        return Component.FIT_CENTER
    }

    override fun getXOffset(): Int {
        return 0
    }

    override fun getYOffset(): Int {
        return -30
    }
}